battle.cpp

```diff
@@ -377,7 +377,7 @@ void AddBattleFact(
 ) {
     if (!Globals->WORLD_EVENTS) return;
 
-    auto fact = new BattleFact;
+    auto fact = new BattleFact();
```

Comment:
*this* is the bug that was causing the news output to randomize if you had any pvp battles.
While this didn't show up in the snapshot tests, it showed up in the regression suite I run using mid-game NO7 turns.